File: LeetCode Solutions/June 2022 LeetCoding Challenge/Intersection of Two Linked Lists.py
# Definition for singly-linked list.
# class ListNode:
#     def __init__(self, x):
#         self.val = x
#         self.next = None

class Solution:
    def getIntersectionNode(self, headA: ListNode, headB: ListNode) -> Optional[ListNode]:
        head1 = headA
        head2 = headB
        while(head1 != head2):
            if(head1 == None):
                head1 = headB
            else:
                head1 = head1.next
            if(head2 == None):
                head2 = headA
            else:
                head2 = head2.next
        return head1

File: LeetCode Solutions/Learn/Linked List/Intersection of Two Linked Lists/Intersection of Two Linked Lists.c
/**
 * Definition for singly-linked list.
 * struct ListNode {
 *     int val;
 *     struct ListNode *next;
 * };
 */
struct ListNode *getIntersectionNode(struct ListNode *headA, struct ListNode *headB) {
    struct ListNode *head1 = headA;
    struct ListNode *head2 = headB;
    while (head1 != head2){
        if (head1 == NULL){
            head1 = headB;
        } else{
            head1 = head1->next;
        }
        if (head2 == NULL){
            head2 = headA;
        } else{
            head2 = head2->next;
        }
    }
    return head1;
}